velox/functions/prestosql/window/tests/WindowTestBase.cpp

```diff
@@ -92,19 +96,29 @@ std::vector<RowVectorPtr> WindowTestBase::makeFuzzVectors(
 void WindowTestBase::testWindowFunction(
     const std::vector<RowVectorPtr>& input,
     const std::string& function,
-    const std::string& overClause) {
-  auto queryInfo = buildWindowQuery(input, function, overClause);
+    const std::string& overClause,
+    const std::string& frameClause) {
+  auto queryInfo = buildWindowQuery(input, function, overClause, frameClause);
   SCOPED_TRACE(queryInfo.functionSql);
   assertQuery(queryInfo.planNode, queryInfo.querySql);
 }
 
 void WindowTestBase::testWindowFunction(
     const std::vector<RowVectorPtr>& input,
     const std::string& function,
-    const std::vector<std::string>& overClauses) {
+    const std::vector<std::string>& overClauses,
+    const std::vector<std::string>& frameClauses) {
```

Comment:
You could change the default value of frameClauses here to be {""}. And then write the loops in the opposite order
for each over clause
   for each frame clause

That would remove the special case here.

Reply:
Thanks for the suggestion, modified accordingly.